redb-derive/tests/derive_tests.rs

```diff
@@ -0,0 +1,227 @@
+use redb::{Database, TableDefinition, Value};
+use redb_derive::Value as DeriveValue;
+use tempfile::NamedTempFile;
+
+#[derive(Debug, DeriveValue)]
+struct SimpleStruct {
+    id: u32,
+    name: String,
+}
+
+#[derive(Debug, DeriveValue)]
+struct TupleStruct(u64, bool);
+
+#[derive(Debug, DeriveValue)]
+struct SingleField {
+    value: i32,
+}
+
+#[derive(Debug, DeriveValue)]
+struct MixedTypes {
+    fixed: u16,
+    variable: String,
+    optional: Option<u8>,
+}
+
+fn create_tempfile() -> NamedTempFile {
+    NamedTempFile::new().unwrap()
+}
+
+#[test]
+fn test_simple_struct_type_name() {
+    assert_eq!(
+        "SimpleStruct {id: u32, name: String}",
+        format!("{}", SimpleStruct::type_name())
+    );
+}
+
+#[test]
+fn test_tuple_struct_type_name() {
+    assert_eq!(
+        "TupleStruct(u64,bool)",
+        format!("{}", TupleStruct::type_name())
+    );
+}
+
+#[test]
+fn test_single_field_type_name() {
+    assert_eq!(
+        "SingleField {value: i32}",
+        format!("{}", SingleField::type_name())
+    );
+}
+
+#[test]
+fn test_mixed_types_type_name() {
+    assert_eq!(
+        "MixedTypes {fixed: u16, variable: String, optional: Option < u8 >}",
```

Comment:
Hmm, the formatting seems weird here. Why is it `Option < u8 >`? I'd like it to be `Option<u8>`